auth/auth.go

```diff
@@ -233,9 +242,33 @@ func (auth *Authenticator) calculateHistory(invalSeq uint64, invalGrants ch.Time
 		currentHistory[previousName] = currentHistoryForGrant
 	}
 
+	if prunedHistory := currentHistory.PruneHistory(auth.ClientPartitionWindow); len(prunedHistory) > 0 {
+		base.Debugf(base.KeyCRUD, "rebuildChannels: Pruned principal history on %s for %s", princName, strings.Join(prunedHistory, ","))
+	}
+
 	return currentHistory
 }
 
+func CalculateMaxHistoryEntriesPerGrant(historyLength int) int {
```

Comment:
It's not immediately obvious what this is.

```suggestion
func CalculateMaxHistoryEntriesPerGrant(numChannels int) int {
```